sw/device/silicon_creator/lib/drivers/flash_ctrl.c

```diff
@@ -208,12 +208,19 @@ static rom_error_t write(uint32_t addr, flash_ctrl_partition_t partition,
  * @return Base address of the given page.
  */
 static uint32_t info_page_addr(flash_ctrl_info_page_t info_page) {
-  const uint32_t bank_index =
-      bitfield_bit32_read(info_page, FLASH_CTRL_INFO_PAGE_BIT_BANK);
-  const uint32_t page_index =
-      bitfield_field32_read(info_page, FLASH_CTRL_INFO_PAGE_FIELD_PAGE);
-  return kMemBase + bank_index * FLASH_CTRL_PARAM_BYTES_PER_BANK +
-         page_index * FLASH_CTRL_PARAM_BYTES_PER_PAGE;
+#define INFO_PAGE_ADDR_CASE_(name_, value_, bank_, page_)       \
+  case name_:                                                   \
+    SHUTDOWN_CHECK(launder32(info_page) == name_);              \
+    return kMemBase + bank_ * FLASH_CTRL_PARAM_BYTES_PER_BANK + \
```

Comment:
I assume this arithmetic operation gets collapsed to a constant by the compiler.

Reply:
I took a quick look at the disassembly, it looks like it is.